types.go

```diff
@@ -239,9 +247,13 @@ type Settings struct {
 // DefaultSettings returns a new Settings struct containing the default settings.
 func DefaultSettings() *Settings {
 	return &Settings{
-		GradientAbsTol: defaultGradientAbsTol,
-		FunctionAbsTol: math.Inf(-1),
-		Recorder:       NewPrinter(),
+		GradientThreshold: defaultGradientAbsTol,
+		FunctionThreshold: math.Inf(-1),
+		Recorder:          NewPrinter(),
+		FunctionConverge: &FunctionConverge{
+			Absolute:   1e-10,
```

Comment:
Can we set also Relative to non-zero value? Magnitude of the minimum is unknown, if it is zero, Absolute will matter, if it is non-zero, Relative will matter. Perhaps 1e-6?